### PR TITLE
v3-2156: Add partner_id on user creation

### PR DESCRIPTION
**Pull request by ethanadams**
What: 
Add partner ID to the user table on user creation, if the partner ID is provided.

Add partner ID to api key if the user creating the API key has a partner ID associated with it.

Why:
Associate a user to a partner ID for value attribution
Associate a api key to a partner ID

Please describe the tests:
 - Test 1: Updated existing users_test
 - Test 2: Updated existing mutation_test
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/users.go [hunk 1]
```diff
@@ -78,6 +82,7 @@ func (users *users) Delete(ctx context.Context, id uuid.UUID) (err error) {
 // Update is a method for updating user entity
 func (users *users) Update(ctx context.Context, user *console.User) (err error) {
```
Comment (coyle) [inline satellite/satellitedb/users.go]: We don't want to allow for the partner_id to be set after the account is created

File: satellite/console/consoleweb/consoleql/user.go [hunk 2]
```diff
@@ -75,6 +77,9 @@ func graphqlUserInput() *graphql.InputObject {
 			FieldPassword: &graphql.InputObjectFieldConfig{
 				Type: graphql.String,
```
Comment (VinozzZ) [inline satellite/console/consoleweb/consoleql/user.go]: Just curious on why we need to add the `PartnerID` in the `UserInput` type but not in the `User` type?

File: satellite/satellitedb/users.go (hunk 1 above)
Comment (VinozzZ) [reply satellite/satellitedb/users.go]: I think the diff here is misleading. The `Update` method is not changed. It's the `userFromDBX` function that's modified

File: satellite/console/consoleweb/consoleql/user.go (hunk 2 above)
Comment (ethanadams) [reply satellite/console/consoleweb/consoleql/user.go]: updated

**Pull request merged**